### ISSUE TITLE
Async/await when running on pre iOS 13 devices

### ISSUE THREAD

**Issue by gabors** [kean/Nuke#526]
Hi @kean 

I'm not sure this is going to be an issue, but I just ran into a problem with a different framework on iOS 12
https://github.com/Alamofire/Alamofire/issues/3529

The app would crash on iOS 12 and earlier because of libswift_Concurrency.dylib even though it's not being used.

Just wanted to bring this up in case we may run into similar problem with Nuke 10.6.
I don't have the ability to test this on a pre iOS 13 device using an AppStoreConnect build as TestFlight is not supported anymore pre 13.

**Comment by kean** [comment]
Oh, wow. Yeah, thanks for bringing this up. If that's the case, I think it would be best to revert the changes for now. I'm not sure I can afford to jump from iOS 11 to iOS 13 as a minimum required version just yet.

**Comment by gabors** [comment]
Yeah we are going the same conundrum now. :)

Xcode 13.2 release notes has a suggested workaround here 
https://developer.apple.com/documentation/xcode-release-notes/xcode-13_2-release-notes

```Apple Clang Compiler
Known Issues
Apps built with Xcode 13 or Xcode 13.1 that make use of Swift Concurrency features (such as async/await), deploy to iOS prior to 15, tvOS prior to 15, or watchOS prior to 8, and have bitcode enabled may crash at launch with an error reporting that the libswift_Concurrency.dylib library was not loaded. (86349088)
Workaround: Add -Wl,-weak-lswift_Concurrency -Wl,-rpath,/usr/lib/swift to Other Linker Flags in the app’s build settings.```

But since it is impossible to test this for me without pushing an app to the AppStore I'm hesitant on using it.

**Comment by kean** [comment]
I deleted the release for now until there is more information. If someone's interested in using async/await, please install Nuke for the branch.

**Comment by gabors** [comment]
Cool, thanks. Probably for the best for the time being.

**Comment by kean** [comment]
С Наступающим!

**Comment by mohpor** [comment]
Alamofire did resolve this. It can be done. Very intereted to have the async/await.

Checkout this: 
https://github.com/Alamofire/Alamofire/issues/3529#issuecomment-1100938606

P.S.: why not have it as extension module for now until we are sure it won't break anything? Those who are iOS 13.0+ can use it at least!

**Comment by kean** [comment]
Thanks, @mohpor. This is fantastic news! I'll release async/await this weekend.

**Comment by gabors** [comment]
@mohpor @kean 
Alamofire did not really change anything to resolve this issue.
They simply made the async/await code require Swift 5.6 and Xcode 13.3 happens to resolve the AppStore connect issue.

**Comment by kean** [comment]
Yes, I did the same. My understanding is that it addresses the issue with iOS 12 and earlier devices.

```swift
// Before:
#if swift(>=5.5.2))

// After
#if swift(>=5.6)
```

**Comment by mohpor** [comment]
@gabors Yes. I know. But, there was a fix in Xcode 13.3.1 that made it possible.

> Fixed: Exporting an app that uses Swift’s concurrency features from an archive with bitcode might fail when the app targets iOS 13.0–14.7, watchOS 6.0–7.6, or tvOS 13.0–14.7. (89271047)

I hope this will resolve some of the issues.

**Comment by kean** [comment]
Async/Await available in version [10.8](https://github.com/kean/Nuke/releases/tag/10.8.0). I tested it on iOS 12 simulator – doesn't crash.
